compiler/src/scala/quoted/runtime/impl/Matcher.scala

```diff
@@ -348,10 +348,21 @@ object Matcher {
      *   - The scrutinee has is in the environment and they are equivalent
      *   - The scrutinee overrides the symbol of the pattern
      */
-    private def symbolMatch(scrutinee: Symbol, pattern: Symbol)(using Env): Boolean =
-      scrutinee == pattern
-      || summon[Env].get(scrutinee).contains(pattern)
-      || scrutinee.allOverriddenSymbols.contains(pattern)
+    private def symbolMatch(scrutineeTree: Tree, patternTree: Tree)(using Env): Boolean =
+      val scrutinee = scrutineeTree.symbol
+      val devirtualizedScrutinee = scrutineeTree match
+        case Select(qual, _) =>
+          // TODO improve performance
+          qual.tpe.typeSymbol.methods.find( sym =>
+            sym.name == scrutinee.name &&
+            sym.allOverriddenSymbols.contains(scrutinee)
+          ).getOrElse(scrutinee)
+        case _ => scrutinee
+      val pattern = patternTree.symbol
+
+      devirtualizedScrutinee == pattern
+      || summon[Env].get(devirtualizedScrutinee).contains(pattern)
+      || devirtualizedScrutinee.allOverriddenSymbols.contains(pattern)
```

Comment:
The document needs to be updated. The following code might be faster:

```Scala
scrutinee.overridingSymbol(pattern.owner) == pattern
|| pattern.overridingSymbol(scrutinee.owner) == scrutinee
```

Reply:
Will try that. The current performance is unaccepable.

Reply:
`overridingSymbol` worked, but not there.

Reply:
If the check is whether the two symbols override each other, can we simply replace the body of the method with the following:

```Scala
scrutinee.overridingSymbol(pattern.owner) == pattern
|| pattern.overridingSymbol(scrutinee.owner) == scrutinee
```

Reply:
That version had some false positives involving modules.